builder/src/build/patches/investigated/update-webpack-chunks-file/get-file-content-with-updated-webpack-require-f-code.ts

```diff
@@ -0,0 +1,74 @@
+import * as ts from "ts-morph";
+
+export async function getFileContentWithUpdatedWebpackFRequireCode(
+  sourceFile: ts.SourceFile,
+  { installChunk, installedChunks }: { installChunk: string; installedChunks: string },
+  chunks: string[]
+): Promise<string> {
+  const webpackFRequireFunction = sourceFile
+    .getDescendantsOfKind(ts.SyntaxKind.BinaryExpression)
+    .map((binaryExpression) => {
+      const binaryExpressionLeft = binaryExpression.getLeft();
+      if (!binaryExpressionLeft.getText().endsWith(".f.require")) return;
+
+      const binaryExpressionOperator = binaryExpression.getOperatorToken();
+      if (binaryExpressionOperator.getText() !== "=") return;
+
+      const binaryExpressionRight = binaryExpression.getRight();
+      const binaryExpressionRightText = binaryExpressionRight.getText();
+
+      if (
+        !binaryExpressionRightText.includes(installChunk) ||
+        !binaryExpressionRightText.includes(installedChunks)
```

Comment:
So `binaryExpressionRightText` is the text of the whole fat arrow function that makes use of the `installChunk` and `installedChunks` identifiers? So we need both to exist in the text of the function.

Reply:
yes exactly 🙂

Reply:
I've introduced a new variable: `functionUsesChunkInstallationVariables` to hopefully make the code clearer 🙂